src/BaseRepository.php

```diff
@@ -122,7 +122,7 @@ public function selectByBoundaries(array $condition = [], array $params = [], in
 	 */
 	public function update(BaseModel $model)
 	{
-		return $this->dba->update(static::$table_name, $model->toArray(), ['id' => $model->id], true);
+		return $this->dba->update(static::$table_name, $model->toArray(), ['id' => $model->id], $model->getOriginalData());
```

Comment:
This is a **real** elegant way for solving this problem!!! nice

Reply:
Thanks, I was pretty proud of myself and I was hoping you would appreciate it as well.

Reply:
btw. It's based on the "Memento" pattern just if you like to read some patterns:
https://designpatternsphp.readthedocs.io/en/latest/Behavioral/Memento/README.html
:-)

Reply:
Ha ha, only loosely!